discovery_node/substrate-ingest/src/joynode/StateBootstrap.ts

```diff
@@ -0,0 +1,58 @@
+import ISubstrateQueryService from './ISubstrateQueryService';
+import { ApiPromise } from '@polkadot/api';
+
+import { IProfile } from '@joystream/types/lib/members';
+import { Hash } from '@polkadot/types/interfaces';
+import { Codec } from '@polkadot/types/types';
+import { EventEmitter } from 'events';
+import Config from '../Config';
+import { JoyStruct } from '@joystream/types/lib/JoyStruct';
+const logger = require('log4js').getLogger('state-boot');
+
+const DEFAULT_BOOTSTRAP_HEIGHT = 0;
+
+export interface JoyEntity {
+    entity: Object,
+    type: string,
+    blockNumber: number,
+    id: any
+}
+
+export default class StateBootstrap extends EventEmitter {
```

Comment:
Probably move it to `QueryNode` then? My thinking was to isolate the initial state bootstrapping (which requires separate and enitity-specific API calls) from the rest of the event ingestion. Since the block producer is one level beneath the query node class, it seemed natural to move this logic into a separate class as well. Perhaps an even cleaner approach is to further extract an interfaces from `QueryNode`.